src/transformers/models/deta/modeling_deta.py

```diff
@@ -1778,7 +1778,7 @@ def forward(
 )
 class DetaForObjectDetection(DetaPreTrainedModel):
     # When using clones, all layers > 0 will be clones, but layer 0 *is* required
-    _keys_to_ignore_on_load_missing = ["bbox_embed\.[1-9]\d*", "class_embed\.[1-9]\d*"]
+    _keys_to_ignore_on_load_missing = ["bbox_embed\.[1-9]\d*", "class_embed\.[1-9]\d*", "model.decoder"]
```

Comment:
`model.decoder` can share `bbox_embed` and `class_embed`.

It's not necessarily the case and depends on the config. 
This is the sort of model which make the dynamic nature of the PR crucial (drop **only** shared tensors).

The debt we're creating here, is that if a file is missing some weights (like model.decoder.bbox_embed) but the config is set to **not** sharing (or modified on disk). Then the warning will not be shown yet keeping random weights on the model.

I don't see any good ways to solve this since `_keys_to_ignore` is not config dependent.
However, since the links are properly made a init time (so when the config is allowed) the issue will only arise when users use mismatched config and weights.

Probably and acceptable choice.